plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergGlueCatalogAccessOperations.java

```diff
@@ -0,0 +1,502 @@
+import static io.trino.plugin.iceberg.TableType.MANIFESTS;
+import static io.trino.plugin.iceberg.TableType.PARTITIONS;
+import static io.trino.plugin.iceberg.TableType.PROPERTIES;
+import static io.trino.plugin.iceberg.TableType.SNAPSHOTS;
+import static io.trino.testing.TestingSession.testSessionBuilder;
+import static io.trino.testing.sql.TestTable.randomTableSuffix;
+import static java.lang.String.format;
+import static java.lang.String.join;
+import static java.lang.annotation.ElementType.FIELD;
+import static java.lang.annotation.ElementType.METHOD;
+import static java.lang.annotation.ElementType.PARAMETER;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+import static java.util.Objects.requireNonNull;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.testng.Assert.fail;
+
+/*
+ * TestIcebergGlueCatalogAccessOperations currently uses AWS Default Credential Provider Chain,
+ * See https://docs.aws.amazon.com/sdk-for-java/v1/developer-guide/credentials.html#credentials-default
+ * on ways to set your AWS credentials which will be needed to run this test.
+ */
+@Test(singleThreaded = true)
+public class TestIcebergGlueCatalogAccessOperations
+        extends AbstractTestQueryFramework
+{
+    private final String schema = "test_schema_" + randomTableSuffix();
+    private final Session TEST_SESSION = testSessionBuilder()
+            .setCatalog("iceberg")
+            .setSchema(schema)
+            .build();
+
+    private GlueMetastoreStats catalogStats;
+    private GlueMetastoreStats tableOperationStats;
+
+    @Override
+    protected QueryRunner createQueryRunner()
+            throws Exception
+    {
+        File tmp = Files.createTempDirectory("test_iceberg").toFile();
+        DistributedQueryRunner queryRunner = DistributedQueryRunner.builder(TEST_SESSION).build();
+
+        AtomicReference<GlueMetastoreStats> catalogStatsReference = new AtomicReference<>();
+        AtomicReference<GlueMetastoreStats> tableOperationStatsReference = new AtomicReference<>();
+
+        queryRunner.installPlugin(new TestingIcebergPlugin(
+                Optional.empty(),
+                Optional.empty(),
+                Optional.of(new StealStatsModule(catalogStatsReference, tableOperationStatsReference))));
+        queryRunner.createCatalog("iceberg", "iceberg",
+                ImmutableMap.of(
+                        "iceberg.catalog.type", "glue",
+                        "hive.metastore.glue.default-warehouse-dir", tmp.getAbsolutePath()));
+
+        queryRunner.execute("CREATE SCHEMA " + schema);
+
+        catalogStats = verifyNotNull(catalogStatsReference.get(), "catalogStatsReference not set");
+        tableOperationStats = verifyNotNull(tableOperationStatsReference.get(), "tableOperationStatsReference not set");
+        return queryRunner;
+    }
+
+    @AfterClass(alwaysRun = true)
+    public void cleanUpSchema()
+    {
+        getQueryRunner().execute("DROP SCHEMA IF EXISTS " + schema);
```

Comment:
IF EXISTS is redundant, since it should exist
(... and can mask problems, like wrong name)

```suggestion
        getQueryRunner().execute("DROP SCHEMA " + schema);
```